.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -208,13 +218,13 @@ jobs:
             --build-config-path "${CONFIG_PATH}" \
             --build-definition-path build-definition.json \
             --builder-image "${BUILDER_IMAGE}@${BUILDER_DIGEST}" \
-            --git-commit-digest "sha1:${GITHUB_SHA}" \
+            --git-commit-digest "sha1:${{needs.commit.outputs.sha}}" \
```

Comment:
Ah, are you having trouble getting `GITHUB_SHA`? Hmmm

Reply:
Yes. In case of pull requests it is not the actual commit. I am trying to get the latest commit, but now I get the error `fatal: not a git repository (or any of the parent directories): .git`. This is very strange.

Reply:
I thought `detect-env` would give the correct sha, but it gives empty string.

Reply:
Ohhh gotcha, for `pull_request` events it does appear `GITHUB_SHA` is not the triggering commit (https://github.com/orgs/community/discussions/26325). But for that, wouldn't the builder need to checkout the base repository (which may be the fork?)

> I thought detect-env would give the correct sha, but it gives empty string.

detect-env gives the ref of the reusable workflow, not the triggering environment. hmmm. It should give the head sha of that, let me see if I can test that out as well.

Reply:
>  But for that, wouldn't the builder need to checkout the base repository (which may be the fork?)

Yes, but even in that case the `GITHUB_SHA` we provide to the command is not something that exists on the repo. 

It should be possible to get the actual commit hash. I have one more idea to try :)

Reply:
I think detect-env has special case for pull requests to enable pre-submit tests. We don't really support it out outside of that context. The action may need to be adjusted.